docs/config/intro.md

```diff
@@ -0,0 +1,59 @@
+Configuration Files
+===================
+
+Forge has built in support for creating and managing mod configuration files. The purpose of a configuration file is to add flexibility and allow end users to customize their experience.
+
+Creating
+--------
+
+Create a dedicated class. Use the builder instance to define the configuration structure and data.
+
+```java
+private BooleanValue mySetting;
+
+public ServerConfig(ForgeConfigSpec.Builder builder) {
+  builder.comment("examplemod settings"); // Comment at the start of the file
```

Comment:
You don't need a comment at the start of the file. You should instead mention that you can call comment before push/pull to add comments above/below categories.

Reply:
Its obvious that a comment isn't required. The first comment is for demonstration.

Reply:
Added text " Comments are 100% optional." for clarity.

Reply:
My problem was rather that the comment is not in fact necessarily at the start of the file, not about them being required or not. The comment before a category should explain what kind of settings there are in the category, someone seeing it here would assume it's a generic header comment above the file whilst it isn't, unless you make one big category at the top.